.github/workflows/experimental-github-actions-testing.yml

```diff
@@ -7,4 +7,10 @@ jobs:
   test_job:
     runs-on: [self-hosted, basic_runner_group]
     steps:
-      - run: echo hi
+      - uses: actions/checkout@v4
+        with:
+          # By default, checkout merges the PR into the current master.
+          # Instead, we want to check out the PR as is.
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Should we be doing this or doing the merge? This behavior matches what we already do in TC, so it might be surprising to change it.